integration_tests/e2e/goal/updateGoal.cy.ts

```diff
@@ -114,6 +114,35 @@ context('Update a goal', () => {
       })
   })
 
+  it('should be able to remove a step as part of updating a goal', () => {
+    // Given
+    const prisonNumber = 'G6115VJ'
+    cy.signIn()
+
+    cy.visit(`/plan/${prisonNumber}/view/overview`)
+    const overviewPage = Page.verifyOnPage(OverviewPage)
+
+    const updateGoalPage = overviewPage.clickUpdateButtonForFirstGoal()
+
+    // When
+    updateGoalPage //
+      .clickRemoveButtonForSecondStep()
+      .submitPage()
+
+    const reviewUpdateGoalPage = Page.verifyOnPage(ReviewUpdateGoalPage)
+    reviewUpdateGoalPage.submitPage()
+
+    // Then
+    Page.verifyOnPage(OverviewPage)
+    // Assert that the expected step was removed from the UpdateGoal request to the API - TODO, there has to be a better way of doing this
+    cy.task<UpdateGoalRequest>('getUpdateGoalRequestBody')
+      .then(updateGoalRequestBody => {
+        const { steps } = updateGoalRequestBody
+        return steps
+      })
+      .should('have.length', 0)
```

Comment:
Does this assertion suggest we are sending the update goal request to the API with zero steps?
That shouldn't be allowed - we must always have at least 1 step

Reply:
And if the When is removing the 2nd step (`clickRemoveButtonForSecondStep()`); then the first step should still be there (plus any steps that were after step 2)